rdtools/system_analysis.py

```diff
@@ -0,0 +1,585 @@
+               default value of 0 neglects the wind in this calculation
+    albedo: Albedo to be used in irradiance transposition calculations (numeric)
+    temperature_model: model parameter pvsystem.sapm_celltemp() used in calculating cell temperature from ambient
+    pv_azimuth: Azimuth of PV array in degrees from north (numeric)
+    pv_tilt: Tilt of PV array in degrees from horizontal
+    pv_nameplate: Nameplate DC rating of PV array in Watts (numeric)
+    interp_freq: The frequency to which to interpolate all Pandas Time Series passed at instantiation. We recommend using the natural
+                 frequency of the data, rather than up or down sampling. Analysis requires regular time series.
+    max_timedelta: The maximum gap in the data to be interpolated/integrated across when interpolating or calculating energy from power (Timedelta)
+    '''
+    
+    def __init__(self, pv, poa=None, temperature=None, temperature_coefficient=None,
+                 aggregation_freq='D', pv_input='power', temperature_input='cell', pvlib_location=None,
+                 clearsky_poa=None, clearsky_temperature=None, clearsky_temperature_input='cell', windspeed=0, albedo=0.25,
+                 temperature_model=None, pv_azimuth=None, pv_tilt=None, pv_nameplate=None, interp_freq=None, max_timedelta=None):
+        '''
+        Instantiates a system_analysis object
+        '''
+        if interp_freq is not None:
+            pv = normalization.interpolate(pv, interp_freq, max_timedelta)
+            if poa is not None:
+                poa = normalization.interpolate(poa, interp_freq, max_timedelta)
+            if temperature is not None:
+                temperature = normalization.interpolate(temperature, interp_freq, max_timedelta)
+            if clearsky_poa is not None:
+                clearsky_poa = normalization.interpolate(clearsky_poa, interp_freq, max_timedelta)
+            if clearsky_temperature is not None:
+                clearsky_temperature = normalization.interpolate(clearsky_temperature, interp_freq, max_timedelta)
+            if isinstance(pv_azimuth, (pd.Series, pd.DataFrame)):
+                pv_azimuth = normalization.interpolate(pv_azimuth, interp_freq, max_timedelta)
+            if isinstance(pv_tilt, (pd.Series, pd.DataFrame)):
+                pv_tilt = normalization.interpolate(pv_tilt, interp_freq, max_timedelta)
+
+        if pv_input == 'power':
+            self.pv_power = pv
+            self.pv_energy = normalization.energy_from_power(pv, max_timedelta=max_timedelta)
+        elif pv_input == 'energy':
+            self.pv_power = None
+            self.pv_energy = pv
+
+        if temperature_input == 'cell':
+            self.cell_temperature = temperature
+            self.ambient_temperature = None
+        elif temperature_input == 'ambient':
+            self.cell_temperature = None
+            self.ambient_temperature = temperature
```

Comment:
The logic in this block above could be simplified a bit by changing the arguments in the init function to be `cell_temperature` and `ambient_temperature`, instead of `temperature` and `temperature_input`. 
you could then just have two lines: 
self.cell_temperature = cell_temperature
self.ambient_temperature = ambient_temperature

Reply:
Separate `cell_temperature` and `ambient_temperature` arguments also avoids the issue of misspelled `'cell'` or `'ambient'` strings passing through the `if/elif` blocks and causing mysterious errors later on.